### PR TITLE
[MNOE-1083] Company Cart System

### PR DESCRIPTION
**Pull request by MAhsenArif**

### CODE REVIEW

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 1]
```diff
@@ -39,40 +39,52 @@ def create
     return render_not_found('Organization') unless organization
 
     subscription = MnoEnterprise::Subscription.new(subscription_update_params)
+    # Note: This needs to change based on rework
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: Is this comment still valid?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 2]
```diff
@@ -39,40 +39,52 @@ def create
...
 
   # PUT /mnoe/jpi/v1/admin/organizations/1/subscriptions/abc
   def update
+    set_staged_subscription_params
     subscription = fetch_subscription(params[:organization_id], params[:id])
     return render_not_found('subscription') unless subscription
     subscription.attributes = subscription_update_params
 
     edit_action = params[:subscription][:edit_action]
+    # TODO: make sure this handles modify_cart!
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: Does it handle modify cart? :trollface:

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/subscriptions_controller.rb [hunk 3]
```diff
@@ -45,14 +46,17 @@ def create
...
 
-    subscription = MnoEnterprise::Subscription.where(organization_id: parent_organization.id, id: params[:id]).first
+    status_params = { organization_id: parent_organization.id, id: params[:id] }
+    status_params[:status_for] = 'staged' if cart_subscription_param.present?
+    subscription = MnoEnterprise::Subscription.where(status_params).first
     return render_not_found('subscription') unless subscription
 
     subscription.attributes = subscription_update_params
     edit_action = params[:subscription][:edit_action]
+    # TODO: We need to adjust this logic for cart request
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/subscriptions_controller.rb]: ?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 4]
```diff
@@ -128,9 +109,19 @@ def fetch_subscriptions(organization_id)
...
+            .apply_query_params(params)
             .with_params(_metadata: { act_as_manager: current_user.id })
             .where(organization_id: organization_id, id: id)
     rel = rel.includes(*includes) if includes.present?
     rel.first
   end
+
+  def set_staged_subscription_params
+    params[:where] ||= {}
+    params[:where][:subscription_status_in] = cart_subscription_param.present? ? 'staged' : 'non_staged'
```
Comment (BrunoChauvet) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: Could we reuse the backend semantic and have the filters `staged` / `visible` ?
https://github.com/maestrano/maestrano-hub/pull/1004/files

File: core/config/locales/templates/dashboard/provisioning/en.yml [hunk 5]
```diff
@@ -123,3 +123,17 @@ en:
             tooltip:
               aborted: Your order was aborted during fulfillment. Please contact your administrator for more information
               external_billing: "This product is billed externally"
+            cart:
+              title: "Your Company Cart"
+              empty: "Please add some items to the cart"
+              add_to_cart: "Add to Cart and continue Shopping"
+              cancel: "Remove Item"
```
Comment (adamaziz15) [inline core/config/locales/templates/dashboard/provisioning/en.yml]: Is this translation used?

**Pull request merged**